web/src/components/upgrade_service/UpgradeService.test.tsx

```diff
@@ -0,0 +1,135 @@
+        <Routes>
+          <Route path="/upgrade-service/*" element={<UpgradeService />} />
+        </Routes>
+      </MemoryRouter>
+    );
+
+    expect(getByText("Checking required steps...")).toBeDefined();
+  });
+
+  describe("Initial state request", () => {
+    let server: SetupServerApi;
+
+    // Override the API url used by the query
+    beforeAll(() => {
+      process.env.API_ENDPOINT = api;
+    });
+
+    afterAll(() => {
+      process.env.API_ENDPOINT = undefined;
+    });
+
+    afterEach(() => {
+      // Remove any handlers added
+      // in individual tests (runtime handlers).
+      server.resetHandlers();
+      server.close();
+    });
+
+    it("We get routed to the config section if the initial request succeeds and the app is configurable", async () => {
+      server = setupServer(
+        http.get(`${api}/upgrade-service/app/${slug}`, () => {
+          return HttpResponse.json({
+            isConfigurable: true,
+            hasPreflight: false,
+          });
+        })
+      );
+      server.listen();
+
+      const { findByTestId } = render(
+        <MemoryRouter initialEntries={[`/upgrade-service/app/${slug}`]}>
+          <Routes>
+            <Route path="/upgrade-service/*" element={<UpgradeService />} />
+          </Routes>
+        </MemoryRouter>
+      );
+
+      await findByTestId("config-area");
+    });
+
+    it("We get routed to the preflight section if the initial request succeeds and the app is not configurable", async () => {
+      server = setupServer(
+        http.get(`${api}/upgrade-service/app/${slug}`, () => {
+          return HttpResponse.json({
+            isConfigurable: false,
+            hasPreflight: true,
+          });
+        })
+      );
+      server.listen();
+
+      const { findByTestId, getByText } = render(
+        <MemoryRouter initialEntries={[`/upgrade-service/app/${slug}`]}>
+          <Routes>
+            <Route path="/upgrade-service/*" element={<UpgradeService />} />
+          </Routes>
+        </MemoryRouter>
+      );
+
+      await findByTestId("preflight-check-area");
+
+      expect(getByText("Back: Config")).toBeDisabled();
+    });
+
+    it("We get routed to the confirm and deploy section if the initial request succeeds and the app is not configurable and doesn't have preflights", async () => {
+      server = setupServer(
+        http.get(`${api}/upgrade-service/app/${slug}`, () => {
+          return HttpResponse.json({
+            isConfigurable: false,
+            hasPreflight: false,
+          });
+        })
+      );
+      server.listen();
+
+      const { findByTestId, getByText } = render(
+        <MemoryRouter initialEntries={[`/upgrade-service/app/${slug}`]}>
+          <Routes>
+            <Route path="/upgrade-service/*" element={<UpgradeService />} />
+          </Routes>
+        </MemoryRouter>
+      );
+
+      await findByTestId("preflight-check-area");
```

Comment:
is this right? should this be available if preflights are not available?

Reply:
Ah nice catch, thanks! This doesn't make sense though, there's something in the tests that's leading to this behaviour because if I `it.only` this test it fails as expected, if I run the whole file it passes 🙃 need to debug this further tomorrow 😩